gateway/ln-gateway/src/lib.rs

```diff
@@ -1723,20 +1723,26 @@ impl Gateway {
         })
     }
 
+    pub async fn select_client_v2(
```

Comment:
Is there a reason we shouldn't use the existing `select_client()` method?

Reply:
Looks pretty much the same to me

Reply:
We want the and anyhow::Result though.